R/tostring.R

```diff
@@ -778,14 +779,40 @@ wrap_string <- function(str, width, collapse = NULL, smart = TRUE) {
   return(ret)
 }
 
+.go_stri_wrap <- function(str, w) {
+  stringi::stri_wrap(str,
+    width = w,
+    normalize = FALSE, # keeps spaces
+    simplify = TRUE, # If FALSE makes it a list with str elements
+    indent = 0
+  )
+}
+
+# help function: Very rare case where the recursion is stuck in a loop
+force_split_words_by <- function(ret, width) {
+  which_exceeded <- which(nchar(ret) > width)
+  ret_tmp <- NULL
+  for (ii in seq_along(ret)) {
+    if (ii %in% which_exceeded) {
+      wrd_i <- ret[ii]
+      init_v <- seq(1, nchar(wrd_i), by = width)
+      end_v <- c(init_v[-1] - 1, nchar(wrd_i))
+      str_v_tmp <- stringi::stri_sub(wrd_i, from = init_v, to = end_v)
+      ret_tmp <- c(ret_tmp, str_v_tmp[!grepl("^\\s+$", str_v_tmp) & nzchar(str_v_tmp)])
```

Comment:
avoiding empties is needed at this stage to avoid spurious spaces appearing (it should not happen too often to get here anyway, only with seriously small widths)